packages/platforms/browser/lib/auto-instrumentation/full-page-load-plugin.ts

```diff
@@ -54,7 +55,21 @@ export class FullPageLoadPlugin implements Plugin<BrowserConfiguration> {
       return
     }
 
-    const span = this.spanFactory.startSpan('[FullPageLoad]', { startTime: 0, parentContext: null })
+    let parentContext: ParentContext | null = null
+
+    const traceparentMetaTag = document.querySelector('meta[name="traceparent"]')
+    if (traceparentMetaTag !== null && traceparentMetaTag.getAttribute('content')) {
+      // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+      const traceparent = traceparentMetaTag.getAttribute('content')!
```

Comment:
is this non-null assertion necessary, based on the logic above?

Reply:
yes, as there is nothing to stop a function returning something different on a subsequent call, so TypeScript makes no assumptions. Maybe there's a nicer way to write this? I recall thinking about doing an assignment in the if statement but wasn't keen on that personally

Reply:
Well, its much nicer with optional chaining but last time I used that I was asked to reverse it, I think due to impact on bundle size

Reply:
okay, that's totally fine then - I was just surprised this could be null at this stage based on the if statement.